lib/BackgroundJob/Tasks/AddMissingImagesTask.php

```diff
@@ -134,14 +133,14 @@ private function addMissingImagesForUser(string $userId, int $model): int {
 	 * @param Folder $folder Folder to recursively search images in
 	 * @return int Number of missing images found
 	 */
-	private function parseUserFolder(int $model, Folder $folder): int {
+	private function parseUserFolder(string $userId, int $model, Folder $folder): int {
 		$insertedImages = 0;
 		$nodes = $this->getPicturesFromFolder($folder);
 		foreach ($nodes as $file) {
 			$this->logDebug('Found ' . $file->getPath());
 
 			$image = new Image();
-			$image->setUser($file->getOwner()->getUid());
```

Comment:
If I take the user here, this is matiasdelellis@cloud-******. is that of the other instance, so use the userId from addMissingImagesForUser().

If it is shared in our instance, we must consider it local, and each user individually.

My only doubts here, is if we can analyze the photos that someone share directly with us. Perhaps it would not be necessary to analyze when the original user does not activate it (or even installed as in this case.), but it would be a very fine discussion. In practice, if someone shares a file, you can use it however you want... :see_no_evil: 

How does the saying ...? if you can see it you can copy it.. :sweat_smile: